File: frontend/lib/api/carbon.ts
import { apiClient } from './client';
import type { SchemaCarbonQuote, SchemaCarbonProductSummary } from './types.gen';

// 碳交易行情（CEA 全国碳配额 / CCER 国家核证自愿减排 / EUA 欧盟碳配额）。
// 碳价为全国统一行情，不分省。
// 实体类型由 OpenAPI 规范生成（与 customers.ts 同模式）。
export type CarbonQuote = SchemaCarbonQuote;
export type CarbonProductSummary = SchemaCarbonProductSummary;

export const getCarbonSummary = () =>
  apiClient.get<{ items: CarbonProductSummary[] }>('/api/v1/carbon/summary');

export const getCarbonQuotes = (params?: { product?: string; days?: number }) =>
  apiClient.get<{ items: CarbonQuote[] }>('/api/v1/carbon/quotes', { params });

export const genCarbonDemo = () =>
  apiClient.post<{ rows: number; message: string }>('/api/v1/carbon/demo-data');

File: backend/internal/handler/carbon.go
package handler

import (
	"net/http"
	"strconv"

	"github.com/gin-gonic/gin"
	"github.com/ptis/backend/internal/db"
	"github.com/rs/zerolog/log"
)

// CarbonHandler 碳交易行情接口（CEA/CCER/EUA）。
type CarbonHandler struct{ repo *db.CarbonRepository }

func NewCarbonHandler(repo *db.CarbonRepository) *CarbonHandler {
	return &CarbonHandler{repo: repo}
}

// Summary GET /api/v1/carbon/summary —— 三个碳产品的最新行情概览。
func (h *CarbonHandler) Summary(c *gin.Context) {
	list, err := h.repo.Summary(c.Request.Context())
	if err != nil {
		log.Error().Err(err).Msg("查询碳行情概览失败")
		c.JSON(http.StatusInternalServerError, gin.H{"error": "操作失败，请稍后重试"})
		return
	}
	c.JSON(http.StatusOK, gin.H{"items": list})
}

// List GET /api/v1/carbon/quotes?product=&days= —— 行情明细。
func (h *CarbonHandler) List(c *gin.Context) {
	days, _ := strconv.Atoi(c.DefaultQuery("days", "180"))
	product := c.Query("product")
	list, err := h.repo.List(c.Request.Context(), product, days)
	if err != nil {
		log.Error().Err(err).Msg("查询碳行情失败")
		c.JSON(http.StatusInternalServerError, gin.H{"error": "操作失败，请稍后重试"})
		return
	}
	c.JSON(http.StatusOK, gin.H{"items": list})
}

// GenerateDemoData POST /api/v1/carbon/demo-data —— 生成 CEA/CCER/EUA 演示行情。
func (h *CarbonHandler) GenerateDemoData(c *gin.Context) {
	n, err := h.repo.GenerateDemo(c.Request.Context())
	if err != nil {
		log.Error().Err(err).Msg("生成碳行情演示数据失败")
		c.JSON(http.StatusInternalServerError, gin.H{"error": "操作失败，请稍后重试"})
		return
	}
	c.JSON(http.StatusOK, gin.H{"rows": n, "message": "已生成碳交易演示行情（CEA/CCER/EUA）"})
}
